osu.Game/Graphics/Containers/Markdown/OsuMarkdownLinkText.cs

```diff
@@ -1,48 +1,63 @@
 // Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
 // See the LICENCE file in the repository root for full licence text.
 
+using System.Collections.Generic;
 using Markdig.Syntax.Inlines;
 using osu.Framework.Allocation;
+using osu.Framework.Graphics;
 using osu.Framework.Graphics.Containers.Markdown;
-using osu.Framework.Graphics.Sprites;
-using osu.Framework.Input.Events;
+using osu.Game.Online.Chat;
 using osu.Game.Overlays;
 
 namespace osu.Game.Graphics.Containers.Markdown
 {
     public class OsuMarkdownLinkText : MarkdownLinkText
     {
-        [Resolved]
-        private OverlayColourProvider colourProvider { get; set; }
+        [Resolved(canBeNull: true)]
+        private OsuGame game { get; set; }
 
-        private SpriteText spriteText;
+        private readonly string text;
+        private readonly string title;
 
         public OsuMarkdownLinkText(string text, LinkInline linkInline)
             : base(text, linkInline)
         {
+            this.text = text;
+            title = linkInline.Title;
         }
 
         [BackgroundDependencyLoader]
         private void load()
         {
-            spriteText.Colour = colourProvider.Light2;
+            var textDrawable = CreateSpriteText().With(t => t.Text = text);
+
+            InternalChildren = new Drawable[]
+            {
+                textDrawable,
+                new OsuMarkdownLinkCompiler(new[] { textDrawable })
+                {
+                    RelativeSizeAxes = Axes.Both,
+                    Action = OnLinkPressed,
+                    TooltipText = title ?? Url,
+                }
+            };
         }
 
-        public override SpriteText CreateSpriteText()
-        {
-            return spriteText = base.CreateSpriteText();
-        }
-
-        protected override bool OnHover(HoverEvent e)
-        {
-            spriteText.Colour = colourProvider.Light1;
```

Comment:
Where does this get applied now? Is that in a follow-up PR? Because currently the parent container applies `DodgerBlue`..

Reply:
Oh, I guess this is part of what `OsuMarkdownLinkCompiler` does.